diff --git a/base/ca/src/org/dogtagpki/server/ca/rest/ProfileService.java b/base/ca/src/org/dogtagpki/server/ca/rest/ProfileService.java
index 0e708f4..5fffd17 100644
--- a/base/ca/src/org/dogtagpki/server/ca/rest/ProfileService.java
+++ b/base/ca/src/org/dogtagpki/server/ca/rest/ProfileService.java
@@ -388,6 +388,11 @@ public class ProfileService extends PKIService implements ProfileResource {
             throw new BadRequestException("Unable to modify profile state: Invalid Profile Id");
         }
 
+        if (action == null) {
+            CMS.debug("modifyProfileState: invalid request. action is null");
+            throw new BadRequestException("Unable to modify profile state: Invalid action");
+        }
+
         if (ps == null) {
             CMS.debug("modifyProfileState: ps is null");
             throw new PKIException("Error modifying profile state.  Profile Service not available");
