diff -ur pki-tks-orig/shared/conf/CS.cfg pki-tks/shared/conf/CS.cfg
--- pki-tks-orig/shared/conf/CS.cfg	2013-11-21 15:05:50.000000000 -0500
+++ pki-tks/shared/conf/CS.cfg	2013-12-15 14:19:37.000000000 -0500
@@ -292,6 +292,8 @@
 selftests.plugin.TKSKnownSessionKey.cardChallenge=#bd#6d#19#85#6e#54#0f#cd
 selftests.plugin.TKSKnownSessionKey.hostChallenge=#77#57#62#e4#5e#23#66#7d
 selftests.plugin.TKSKnownSessionKey.keyName=#01#01
+selftests.plugin.TKSKnownSessionKey.nistSP800-108KdfOnKeyVersion=FF
+selftests.plugin.TKSKnownSessionKey.nistSP800-108KdfUseCuidAsKdd=false
 selftests.plugin.TKSKnownSessionKey.macKey=#40#41#42#43#44#45#46#47#48#49#4a#4b#4c#4d#4e#4f
 selftests.plugin.TKSKnownSessionKey.sessionKey=#d1#be#b8#26#dc#56#20#25#8c#93#e7#de#f0#ab#4f#5b
 selftests.plugin.TKSKnownSessionKey.token=Internal Key Storage Token
@@ -328,6 +330,8 @@
 tks.defKeySet.auth_key=#40#41#42#43#44#45#46#47#48#49#4a#4b#4c#4d#4e#4f
 tks.defKeySet.mac_key=#40#41#42#43#44#45#46#47#48#49#4a#4b#4c#4d#4e#4f
 tks.defKeySet.kek_key=#40#41#42#43#44#45#46#47#48#49#4a#4b#4c#4d#4e#4f
+tks.defKeySet.nistSP800-108KdfOnKeyVersion=00
+tks.defKeySet.nistSP800-108KdfUseCuidAsKdd=false
 tks.jForte._000=##
 tks.jForte._001=## SAFLink's jForte default key set:
 tks.jForte._002=##
@@ -336,6 +340,8 @@
 tks.jForte.auth_key=#30#31#32#33#34#35#36#37#38#39#3a#3b#3c#3d#3e#3f
 tks.jForte.mac_key=#40#41#42#43#44#45#46#47#48#49#4a#4b#4c#4d#4e#4f
 tks.jForte.kek_key=#50#51#52#53#54#55#56#57#58#59#5a#5b#5c#5d#5e#5f
+tks.jForte.nistSP800-108KdfOnKeyVersion=00
+tks.jForte.nistSP800-108KdfUseCuidAsKdd=false
 multiroles._000=##
 multiroles._001=## multiroles
 multiroles._002=##
